buffer/src/main/java/io/netty5/buffer/internal/CleanerPool.java

```diff
@@ -0,0 +1,200 @@
+
+    /**
+     * Method handle used to optionally load Virtual ThreadFactory if the current platform is supporting Loom.
+     */
+    private static final MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
+
+    /**
+     * Method handle for Thread.ofVirtual() method. Null in case jvm version is < 19
+     */
+    private static final MethodHandle mhOfVirtual;
+
+    /**
+     * Method handle for java.lang.Thread.Builder.OfVirtual.factory() method. Null in case jvm version is < 19
+     */
+    private static final MethodHandle mhFactory;
+
+    /**
+     * Cleaner instances shared by all event loops.
+     */
+    private final Cleaner[] cleaners = IntStream.range(0, POOL_SIZE).mapToObj(i -> createCleaner())
+            .toArray(Cleaner[]::new);
+
+    /**
+     * A FastThreadLocal which allows to map a random cleaner to each thread calling the nextCleaner method.
+     * Cleaners are distributed in round robin, and the same EventLoop will get the same Cleaner for every
+     * nextCleaner method invocations.
+     */
+    private final CleanerThreadLocal threadLocalCleaner = new CleanerThreadLocal(cleaners);
+
+    /**
+     * Static initializer used to setup Method Handles that are used to obtain loom virtual Threadfactory
+     * (if the platform supports it).
+     */
+    static {
+        MethodHandle mhOfVirtualTmp = null;
+        MethodHandle mhFactoryTmp = null;
+
+        if (USE_VTHREAD) {
+            try {
+                Class<?> clzOfVirtual = Class.forName("java.lang.Thread$Builder$OfVirtual");
+                mhOfVirtualTmp = publicLookup.findStatic(Thread.class, "ofVirtual",
+                        MethodType.methodType(clzOfVirtual));
+                mhFactoryTmp = publicLookup.findVirtual(clzOfVirtual, "factory",
+                        MethodType.methodType(ThreadFactory.class));
+            } catch (ClassNotFoundException e) {
+                logger.debug("Loom not supported, Cleaner will use default cleaner daemon threads.");
+            } catch (Throwable t) {
+                logger.warn("Could not create virtual thread factory, will use default cleaner daemon threads.", t);
+            }
+        }
+
+        mhOfVirtual = mhOfVirtualTmp;
+        mhFactory = mhFactoryTmp;
+    }
+
+    /**
+     * The singleton for the CleanerPool.
+     */
+    public static final CleanerPool INSTANCE = new CleanerPool();
+
+    /**
+     * The FastThreadLocal used to map a Cleaner instance to Event Loop threads.
+     */
+    private static class CleanerThreadLocal extends FastThreadLocal<Cleaner> {
+        private final Cleaner[] cleaners;
+        private static final AtomicInteger counter = new AtomicInteger();
+
+        CleanerThreadLocal(Cleaner[] cleaners) {
+            this.cleaners = cleaners;
+        }
+
+        @Override
+        protected Cleaner initialValue() {
+            return cleaners[counter.getAndIncrement() % POOL_SIZE];
```

Comment:
counter could overflow in theory so I think we should better use `abs` to be 100 % safe 
```suggestion
            return cleaners[Math.abs(counter.getAndIncrement() % POOL_SIZE)];
```

Reply:
But if it will be Integer.MIN_VALUE then `Math.abs` will not help. It still will return negative value.

Reply:
hello,
i can suggest the following to avoid negative values:
```suggestion
            return cleaners[(counter.getAndIncrement() & 0x7F_FF_FF_FF) % POOL_SIZE];
```

Reply:
Thanks a lot for the feedbacks. I was thinking to use _Math.floorMod_ or use a pool size which is a power of 2, then use _counter.getAndIncrement() & (POOL_SIZE - 1))_, but I have applied the last suggestion made by @gtixfr (thanks !). 
Please check last commit.